chef-config/lib/chef-config/windows.rb

```diff
@@ -16,13 +16,17 @@
 #
 
 module ChefConfig
-
   def self.windows?
-    if RUBY_PLATFORM =~ /mswin|mingw|windows/
+    if @node
+      platform = @node[:platform] || @node[:platform_family] || @node[:os]
```

Comment:
i don't think you need to do quite this much fallback, on windows you should always have node[:os] and it should always be "windows"

i probably would have shortened this all up to:

```
  def self.windows?
    if @node && @node["os"]
      @node["os"] == "windows"
    else
      RUBY_PLATFORM =~ /mswin|mingw|windows/
    end
end